### PR TITLE
[2/N] Fixes clang-tidy warnings in header files

### PR DESCRIPTION
**Pull request by cyyever**
This PR fixes more clang-tidy warnings in common headers.

### CODE REVIEW

File: torch/csrc/jit/serialization/storage_context.h [hunk 1]
```diff
@@ -62,7 +62,7 @@ class TORCH_API DeserializationStorageContext {
       const DeserializationStorageContext&) = delete;
   DeserializationStorageContext(const DeserializationStorageContext&) = delete;
 
-  void addStorage(const std::string& name, c10::Storage storage) {
+  void addStorage(const std::string& name, const c10::Storage& storage) {
     TORCH_INTERNAL_ASSERT(!hasStorage(name));
     name_storage_map_.insert({name, storage});
```
Comment (Skylion007) [inline torch/csrc/jit/serialization/storage_context.h:67]: Alternatively, std::move here?

**Review by Skylion007 — APPROVED** [review]

File: torch/csrc/jit/serialization/storage_context.h (hunk 1 above)
Comment (cyyever) [reply torch/csrc/jit/serialization/storage_context.h:67]: Move and emplace

**Pull request closed without merging**